### PR TITLE
Tdd milestone

### PR DESCRIPTION
**Pull request by AtillaTahak**
Create a class called Solver.
Create a method called [factorial](https://en.wikipedia.org/wiki/Factorial) that takes one argument, an integer N, and returns the factorial for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
When N is divisible by 3, return "fizz".
When N is divisible by 5, return "buzz".
When N is divisible by 3 and 5, return "fizzbuzz".
Any other case, return N as a string (e.g. say N is 7 then return "7").
Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD

### CODE REVIEW

File: spec/factorial_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require '../solver'
```
Comment (GzimAsani) [inline spec/factorial_spec.rb]: - [ ] Instead of `require` please add `require_relative`

File: spec/fizzbuzz_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require '../solver'
```
Comment (GzimAsani) [inline spec/fizzbuzz_spec.rb]: - [ ] Instead of `require` please add `require_relative`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @AtillaTahak and @Fikerte-T               
Good job so far! You are almost done :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by er-contreras — APPROVED** [review]
Hi @AtillaTahak, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- Well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**